### PR TITLE
fix: make diagnosis filter use any

### PR DESCRIPTION
**Pull request by chinook25**
how to test:
- #3578 

todo:
- [ ] updated docs in case of new feature
- [ ] added tests

### CODE REVIEW

File: apps/directory/src/functions/applyFiltersToQuery.js [hunk 1]
```diff
@@ -99,8 +99,8 @@ export async function applyFiltersToQuery(
         break;
       }
       case "OntologyFilter": {
-        const values = filterValue.map((fv) => fv.code);
-        baseQuery.filter(filterDetail.applyToColumn).in(values);
+        const values = filterValue.map((filterValue) => filterValue.code);
+        baseQuery.orFilter(filterDetail.applyToColumn).in(values);
```
Comment (chinook25) [inline apps/directory/src/functions/applyFiltersToQuery.js:103]: note: this is the actual fix

File: apps/directory/src/components/filters/OntologyFilter.vue [hunk 2]
```diff
@@ -22,11 +22,12 @@
     <div class="ontology pt-3 d-flex justify-content-center">
       <template v-for="ontologyId of ontologyIdentifiers" :key="ontologyId">
         <div v-show="selectedOntology === ontologyId">
-          <spinner class="mt-4 mb-5" v-if="!ontologyOptions" />
+          <spinner class="mt-4 mb-5" v-if="!resolvedOptions" />
           <tree-component
             v-else-if="displayOptions.length"
             :options="displayOptions"
             :filter="ontologyQuery"
+            :facetIdentifier="facetIdentifier"
```
Comment (chinook25) [inline apps/directory/src/components/filters/OntologyFilter.vue:30]: not having this property here caused console errors

Comment (chinook25) [reply apps/directory/src/components/filters/OntologyFilter.vue:30]: This fixes an issue where selecting a diagnosis filter throws  `this.filterInfoDictionary[t]` is undefined in the console

**Review by mswertz — COMMENTED** [review]
code looks good. When functionally approved you can merge.

**Comment by mswertz** [comment]
preview fails?

File: apps/directory/src/stores/biobanksStore.js [hunk 3]
```diff
@@ -120,10 +120,10 @@ export const useBiobanksStore = defineStore("biobanksStore", () => {
         let biobanksWithCollections = biobankResult.Biobanks;
         if (filtersStore.hasActiveFilters) {
           biobanksWithCollections = biobanksWithCollections.filter(
-            (biobank) => biobank.collections
+            (biobank) => biobank.collections?.length
           );
         }
-        biobankCards.value = filterWithdrawn(biobankResult.Biobanks);
```
Comment (chinook25) [inline apps/directory/src/stores/biobanksStore.js:126]: I think this fixes #3577

**Review by konstantina-gkp — APPROVED** [review]
The filter works good. Maybe Esther and Dieuwke should also have a look before merging.

File: apps/directory/src/stores/biobanksStore.js (hunk 3 above)
Comment (dtroelofsprins) [reply apps/directory/src/stores/biobanksStore.js:126]: This was indeed something that was adjusted in the race condition PR: #3562

**Review by dtroelofsprins — APPROVED** [review]
Counts of the biobanks now goes OK as well.

**Pull request merged**